### DISCUSSION TITLE
Finding a similar example in non-Opti syntax

### DISCUSSION THREAD

**Discussion by CheeGY** [casadi/casadi#3800] [Q&A] [answered]
Hi everyone!

Does anyone knows if there is a version of this example in https://web.casadi.org/blog/nlp_sens/, that uses a _non-Opti_ syntax?

In particular, how do I create the functions `M`, `Z`, and subsequently `Z_taylor`, without using `opti.to_function`? 

How do I create similar MX functions if I were to create the solver using syntax along the lines of `solver = nlpsol('solver', 'ipopt', prob, opts)`?

I'm looking for something in Python, but if there is one in MATLAB, that's good too.

Thanks!

**Comment by tmmsartor** [comment, marked as answer]
I replicated the example in the blog without using `opti` (in Python) [here](https://github.com/tmmsartor/casadi_issues/blob/main/3800_nlp_sens_non_opti/main.py), check if this can be useful for you.

You may also want to take a look to the casadi docs example [nlp_sensitivities.py](https://github.com/casadi/casadi/blob/main/docs/examples/python/nlp_sensitivities.py) which showcase additional functionalities exposed to the python interface.

**Reply by CheeGY** [reply]
Thanks! I just ran it and it is working, after changing some of the DM variables to Python objects that matplotlib accepts using `full()`.

1. In Line 39, `plt.plot(pvec.full(),z(S),'o',color='black',markersize=5)`
2. In Line 60, `plt.plot(pvec.full(), (F+J*(pvec-p0)+1/2*H*(pvec-p0)**2).full(), '-', color=color)`
